chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt

```diff
@@ -170,6 +172,48 @@ class RoomSpek : Spek({
 
             assertThat(badEvents).isEmpty()
         }
+
+        it("always has the correct memberUserIds after room data is changed") {
+
+            setUpInstanceWith(createDefaultRole(), newUsers(PUSHERINO, ALICE), newRoom(GENERAL, ALICE))
+            val superUser = chatFor(SUPER_USER).connect().assumeSuccess()
+            val roomUpdated = CountDownLatch(1)
+
+            superUser.subscribeToRoomMultipart(superUser.generalRoom) { event ->
+                when (event) {
+                    is RoomEvent.RoomUpdated -> {
+                        roomUpdated.countDown()
+                    }
+                }
+            }
+
+            assertThat(superUser.generalRoom.memberUserIds.size).isEqualTo(2) //alice and super user
+            superUser.updateRoom(superUser.generalRoom, customData = mapOf("key" to "data") ).assumeSuccess()
+            roomUpdated.await()
+            assertThat(superUser.generalRoom.memberUserIds.size).isEqualTo(2)
```

Comment:
I think we should also assert here that the change in custom data has taken effect (or we should check that such other test cases exist)

Reply:
As well as using `superUser.generalRoom` (which is a test helper for looking the room up via `superUser.rooms`), the user also receives a room entity on the `RoomUpdated` event object - we should also be asserting that the member list is correct on that entity as well.

Reply:
Good point! I've added an assertion in the callback to check the returned event room is as expected, and added the same assertion in the other test we wrote too.

There's already a test for updated room data in the `adds room customData` test

Reply:
Given the two ways of accessing rooms (attached to event vs lookup in the store), it would be good to check that those exists tests validate the new values via both routes.
And because there are so many cases there, it might also be a good candidate for unit testing of this function rather than loads of integration tests.

Reply:
Have updated the previous `adds room customData` to check both routes of accessing the properties (in the callback, and part of the user object). 

Do you think I should look at updating any of the other tests to do this too?

Reply:
NB: Covered this on away from the PR comments. Summary: The other tests use magic `connectFor` and `subscribeFor` to manage their concurrency flow, so they are correct.